### PR TITLE
Add lint-staged and husky

### PR DESCRIPTION
**Pull request by sirugh**
## PR Checklist

- [x] I have read the [Contributing Guidelines](https://github.com/cannawen/metric_units_reddit_bot/blob/master/CONTRIBUTING.md)
- [x] New functionality (or bug fix) is appropriately tested

## Brief description of changes:

I've added [husky](https://github.com/typicode/husky) and [lint-staged](https://github.com/okonet/lint-staged) along with an additional rule that eliminates the need for the `pre-commit` file.

## Verification

1. Check out the PR.
2. Inside a test, make one `it.only` and add a `console.log`:
    ```js
   context('Good bot', () => {
      it.only('should reply', () => {
        console.log('Hello!');
        verify("good bot", "Good human");
        verify("You are such a good robot", "Good human");
      });
    });
    ```

3. Save it and try to commit it. You should be unable due to the failing precommit hook.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,11 +26,18 @@
...
   },
   "repository": {
     "type": "git",
     "url": "https://github.com/cannawen/metric_units_reddit_bot.git"
   },
-  "license": "GPL-3.0"
+  "license": "GPL-3.0",
+  "lint-staged": {
+    "*.js": [
+      "eslint --fix",
```
Comment (sirugh) [inline package.json:39]: This actually auto-fixes whatever it can during a precommit which is really nice!

**Comment by cannawen** [comment]
Thank you for the PR!! Husky looks great :)

I'm a bit worried of adding the "eslint --fix" into the project without going through and immediately fixing all the issues though (#60), because there might be files failing the lint checks and then whoever touches the file next will be forced to fix it. I'm not sure if this a valid is a concern though, will the link checks be generally easy/quick fixes that "anyone" should know how to do? Or we can document somewhere that they can use --no-verify if they don't know how to fix a lint issue (this is only for when #60 is not merged yet. After #60 is merged, I am ok with forcing people to pass the lint check, since it will only be their own code they are touching)

But I am very excited to get eslint in... this inconsistent formatting has been driving me crazy

**Comment by sirugh** [comment]
`eslint --fix` is actually great for the trivial things like ensuring semicolons or adding spaces between function names and parens. You'll probably want to run the entire project through the linter as well to ensure forward compliance as you pointed out. If you have concerns about contributors not knowing how to fix a linting issue I would recommend putting a linting section in your contributing docs that links to the [eslint rules](https://eslint.org/docs/rules/).

As for lint errors/warnings, you can customize the levels to your desired outcome. For example, you probably want the mocha `.only` to error, but an unnamed function can just be a warning.

**Comment by cannawen** [comment]
I checked out your branch, did an npm install, but was still able to commit failing tests, console logs, and "only" statements. Any ideas what is going wrong? I just did a `git add -am "message"` without `--no-verify` ... I got rid of all files in the `./.git/hooks` directory so there's no interference.

Also: the es-lint didn't fix any formatting... very strange

Link to branch: https://github.com/cannawen/metric_units_reddit_bot/commits/sirugh-precommit

**Comment by sirugh** [comment]
Did you see husky run at all? May be related to existing hooks.

**Review by cannawen — CHANGES_REQUESTED** [review]
Disregard previous comment, I hadn't done `npm install` so it was letting me commit!

I had to upgrade my node version because my node version didn't support trailing commas that eslint added... can you update the package.json to get the minimum node version to 8.0.0? Other than that this PR looks amazing, and will help keep our code a lot cleaner, thanks so much!!

**Comment by sirugh** [comment]
Done.

**Comment by cannawen** [comment]
Hey @sirugh can you do a rebase or merge off master? There are conflicts :( I promise I'll merge your PR in next so you don't need to deal with any more conflicts :P

**Review by cannawen — APPROVED** [review]

**Comment by sirugh** [comment]
@cannawen rebased!

**Pull request merged**